src/mds/MDSRank.cc

```diff
@@ -3789,6 +3792,35 @@ void MDSRankDispatcher::handle_conf_change(const ConfigProxy& conf, const std::s
   if (changed.count("mds_enable_op_tracker")) {
     op_tracker.set_tracking(conf->mds_enable_op_tracker);
   }
+  if (changed.count("mds_extraordinary_events_dump_interval")) {
+    reset_event_flags();
+    extraordinary_events_dump_interval = conf.get_val<std::chrono::seconds>
+      ("mds_extraordinary_events_dump_interval").count();
+
+    uint64_t log_level, gather_level;
+    std::string debug_mds = g_conf().get_val<std::string>("debug_mds");
+    auto delim = debug_mds.find("/");
+    std::istringstream(debug_mds.substr(0, delim)) >> log_level;
+    std::istringstream(debug_mds.substr(delim + 1)) >> gather_level;
+
+    //Enable the logging only during low level debugging
+    if (extraordinary_events_dump_interval) {
+      if (log_level < 10) {
+        if (gather_level < 10) {
+          //if gather level is < 10, bump the gather level (>=10) to enable gathering in-memory logs
+          g_conf().set_val("debug_mds", static_cast<std::ostringstream&&>
+                           (std::ostringstream{} << log_level << '/' << 10).str());
```

Comment:
We need to talk about this more.

Reply:
@gregsfortytwo 
Please see: https://github.com/ceph/ceph/pull/44710#pullrequestreview-943090215

Reply:
Pushed 2 new commits for dropping the log level change, while waiting for @lxbsz 's reply.

Reply:
> maybe set the loglevel to 0 conditionally instead ?

How does that solve the issue discussed here (Users expect to control their log levels, and we shouldn't override it.)? I don't understand. We were talking about avoid setting "gather level" from the code.

I think the solution is let the user set the appropriate log_level/gather_level and set the 'mds_extraordinary_events_dump_interval'. Then we will check the log_level/gather_level and schedule the in-memory log dump from the code. That resolves the 'overriding' issue mentioned by @gregsfortytwo .

Reply:
> > maybe set the loglevel to 0 conditionally instead ?
> 
> How does that solve the issue discussed here (Users expect to control their log levels, and we shouldn't override it.)? I don't understand. We were talking about avoid setting "gather level" from the code.

Sorry for confusing. I mean something like:

```
if (...) {
  dout(0) <<...
} else {
  dout(10) <<...
}
```

> 
> I think the solution is let the user set the appropriate log_level/gather_level and set the 'mds_extraordinary_events_dump_interval'. Then we will check the log_level/gather_level and schedule the in-memory log dump from the code. That resolves the 'overriding' issue mentioned by @gregsfortytwo .

Reply:
@lxbsz Considering Greg's and Venky's inputs, I think it's much better let the user set the `debug_mds` values prior to enabling the log dump. I've updated the code and the docs accordingly.

As we've agreed earlier, we don't want to dump in-memory logs for log level >=10, as it is high log level. So a log level = 10 would make it disabled.

Reply:
Checked it and the new changes look good to me.